src/pj_gc_reader.c

```diff
@@ -210,7 +223,7 @@ static int pj_gc_readentry(projCtx ctx, PAFile fid, PJ_GridCatalogEntry *entry)
             entry->date = pj_gc_parsedate( ctx, tokens[6] );
     }
 
-    for( i = 0; i < token_count; i++ )
+    for( i = (token_count < 5) ? 0 : 1; i < token_count; i++ )
```

Comment:
A comment would be helpful here. I agree that given the structure of the existing code, your solution takes the right approach, by eliminating a strdup. Making it optimally clear would mean a major restructuring.

But perhaps a slightly more clear way (and an alternative to a comment) would be to duplicate the deallocation inside the `if (token_count < 5)` branch, returning `1` from inside the `if`. Then deleting the `else` line and let the code block  do its thing, knowing that everything is OK.  This would eliminate the ternary assignment to `i` inside the `for` loop, and only explicitly refer to the boundary value that is relevant in each separate case of failure vs. success.

Reply:
I agree, that is pretty ugly. I think a better solution is to set `tokens[0] = NULL`, because then [free](https://linux.die.net/man/3/free) is a no-op.